Sources/UtiliKit/ActiveLabel/ActiveLabel.swift

```diff
@@ -0,0 +1,285 @@
+     */
+    override var text: String? {
+        didSet {
+            textDidUpdate()
+        }
+    }
+    
+    override var isHidden: Bool {
+        didSet {
+            if isHidden {
+                hideLoadingViews()
+            } else if !isHidden && isLoading {
+                showLoadingViews()
+            }
+        }
+    }
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        defer {
+            textDidUpdate()
+        }
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        
+        defer {
+            textDidUpdate()
+        }
+    }
+    
+    convenience init(frame: CGRect, configuration: ActiveLabelConfiguration) {
+        self.init(frame: frame)
+        
+        self.estimatedNumberOfLines = configuration.estimatedNumberOfLines
+        self.finalLineTrailingInset = configuration.finalLineTrailingInset
+        self.finalLineLength = configuration.finalLineTrailingInset
+        self.loadingViewColor = configuration.loadingViewColor
+        self.loadingLineHeight = configuration.loadingLineHeight
+        self.loadingLineVerticalSpacing = configuration.loadingLineVerticalSpacing
+        self.loadingAnimationDuration = configuration.loadingAnimationDuration
+        self.loadingAnimationDelay = configuration.loadingAnimationDelay
+        
+        configurationChanged()
+    }
+    
+    override func layoutSubviews() {
+        super.layoutSubviews()
+        
+        loadingViews.forEach { [weak self] (view) in
+            guard let self = self else { return }
+            if let maskLayer = view.layer.mask {
+                maskLayer.bounds = CGRect(x: 0, y: 0, width: self.bounds.width + (maskOffset * 2), height: self.bounds.height)
+                maskLayer.frame = CGRect(x: -maskOffset, y: 0, width: self.bounds.width + (maskOffset * 2), height: self.bounds.height)
+            }
+        }
+    }
+    
+    override func prepareForInterfaceBuilder() {
+        super.prepareForInterfaceBuilder()
+        
+        isDisplayingInStoryboard = true
+        isLoading = showLoadingViewsInStoryboard
+    }
+    
+    /**
+     You must call this function to reset the loading views after you have changed a configuration value.
+     */
+    func configurationChanged() {
+        loadingViews.forEach({ $0.removeFromSuperview() })
+        loadingViews.removeAll()
+        
+        if isLoading {
+            showLoadingViews()
+        }
+    }
+}
+
+private extension ActiveLabel {
+    func textDidUpdate() {
+        isLoading = text == nil
+    }
+    
+    // MARK: - Loading View Functions
+    func showLoadingViews() {
+        if loadingViews.isEmpty {
+            configureLoadingViews()
+        }
+        loadingViews.forEach({ $0.isHidden = false })
+    }
+    
+    func hideLoadingViews() {
+        loadingViews.forEach({ $0.isHidden = true })
+    }
+    
+    func configureLoadingViews() {
```

Comment:
Since this is a long method, it might be good to break it up into a couple of methods (view creation vs constraint configuration) and/or add some comments (especially around the index checks towards the bottom).